pkg/capi2mapi/aws_test.go

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2024 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package capi2mapi
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"k8s.io/utils/ptr"
+	capav1 "sigs.k8s.io/cluster-api-provider-aws/v2/api/v1beta2"
+	capiv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+)
+
+var _ = Describe("capi2mapi AWS", Ordered, func() {
+
+	awsMachineTemplate := &capav1.AWSMachineTemplate{
```

Comment:
I think we need to exhaust this template and populate as many fields as we can convert, and then also add some negative cases around the lossy versions.

Would also be good to have a fully populated expected spec, rather than just checking against nil

Reply:
Yes that's for sure!
Locally I've been running roundtrip conversion tests against real world MachineSets which include fully populated specs. I think we need to get that into a repeatable pipeline for presubmits. So we can continually test against those.

Reply:
Lets talk about this separately, but, I think we should try to get some automated testing set up around this, likely will have to be done in a private repository somehow